src/jit/morph.cpp

```diff
@@ -11416,6 +11416,20 @@ GenTreePtr Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* mac)
                 }
             }
 
+            // If gtOp1 is a GT_FIELD, we need to pass down the mac if
+            // its parent is GT_ADDR, since the address of the field
+            // is part of an ongoing address computation. Otherwise
+            // op1 represents the value of the field and so any address
+            // calculations it does are in a new context.
+            if ((op1->gtOper == GT_FIELD) && (tree->gtOper != GT_ADDR))
```

Comment:
If you move this up above the `if (tree->gtOper == GT_ADD && subMac1 != nullptr)` check on line 11393, we could avoid doing that work when a `GT_FIELD` feeds a `GT_ADD`.